### PR TITLE
Security start

### PR DESCRIPTION
**Pull request by qreqit**

### CODE REVIEW

File: src/main/java/mate/academy/springbootwebgreqit/exception/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -41,4 +42,10 @@ private String getErrorMessage(ObjectError objectError) {
         }
         return objectError.getObjectName();
     }
+
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<String> handleRegistrationException(RegistrationException ex) {
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(ex.getMessage());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/exception/GlobalExceptionHandler.java:48]:
```suggestion
        return ResponseEntity.status(HttpStatus.CONFLICT).body(ex.getMessage());
```

File: src/main/java/mate/academy/springbootwebgreqit/mapper/UserMapper.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package mate.academy.springbootwebgreqit.mapper;
+
+import mate.academy.springbootwebgreqit.dto.user.UserResponseDto;
+import mate.academy.springbootwebgreqit.model.User;
+import org.mapstruct.Mapper;
+import org.mapstruct.MapperConfig;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(componentModel = "spring")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/mapper/UserMapper.java:9]:
```suggestion
@Mapper(componentModel = "spring",
        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
        injectionStrategy = InjectionStrategy.CONSTRUCTOR)
```

File: src/main/java/mate/academy/springbootwebgreqit/model/User.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Entity
+@RequiredArgsConstructor
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/model/User.java:11]:
```suggestion
```

File: src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("Can't register user");
+        }
+        if (!requestDto.getPassword().equals(requestDto.getRepeatPassword())) {
+            throw new RegistrationException("Passwords do not match");
+        }
+        User user = new User();
+        user.setEmail(requestDto.getEmail());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java:27]: it should be mapper logic

File: src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java:20]:
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```

File: src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            throw new RegistrationException("Can't register user");
+        }
+        if (!requestDto.getPassword().equals(requestDto.getRepeatPassword())) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java:23]: remove

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

File: src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.existsByEmail(requestDto.getEmail())) {
+            throw new RegistrationException("Can't register user");
+        }
+        User user = new User();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java:23]:
```suggestion
        User user = userMapper.toUser(requestDto);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one comment

File: src/main/java/mate/academy/springbootwebgreqit/mapper/UserMapper.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import org.mapstruct.factory.Mappers;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@Mapper(componentModel = "spring",
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR)
+public interface UserMapper {
+    UserResponseDto toDto(User user);
+
+    User toUser(UserRegistrationRequestDto responseEntityExceptionHandler);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/mapper/UserMapper.java:19]: responseEntityExceptionHandler ???

File: src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.existsByEmail(requestDto.getEmail())) {
+            throw new RegistrationException("Can't register user");
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/service/impl/UserServiceImpl.java:21]: add email to exception message

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see pr

File: src/main/java/mate/academy/springbootwebgreqit/mapper/UserMapper.java (hunk 8 above)
Comment (qreqit) [reply src/main/java/mate/academy/springbootwebgreqit/mapper/UserMapper.java:19]: ohh

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**